packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -925,6 +925,13 @@ export function getUsageOverride(
   return getScopedDecoratorData(context, usageKey, entity);
 }
 
+export function getDefaultUsage(context: TCGCContext, namespace?: Namespace): UsageFlags {
+  if (!namespace) {
+    return UsageFlags.None;
+  }
+  return getScopedDecoratorData(context, usageKey, namespace) ?? UsageFlags.None;
+}
+
 export function getUsage(context: TCGCContext, entity: Model | Enum): UsageFlags {
```

Comment:
found another logic that we forgot. for union, since it could be union as enum, then it could also have usage.